common/src/main/java/com/wynntils/core/features/FeatureRegistry.java

```diff
@@ -4,66 +4,127 @@
-import com.wynntils.features.ItemHighlightFeature;
-import com.wynntils.features.ItemScreenshotFeature;
-import com.wynntils.features.ItemStatInfoFeature;
-import com.wynntils.features.LootrunFeature;
-import com.wynntils.features.MountHorseHotkeyFeature;
-import com.wynntils.features.MythicBlockerFeature;
-import com.wynntils.features.PlayerGhostTransparencyFeature;
-import com.wynntils.features.SoulPointTimerFeature;
-import com.wynntils.features.WynncraftButtonFeature;
+import com.wynntils.core.features.properties.EventListener;
+import com.wynntils.core.features.properties.KeyBinds;
+import com.wynntils.core.features.properties.StartEnabled;
+import com.wynntils.core.keybinds.KeyHolder;
 import com.wynntils.features.debug.ConnectionProgressFeature;
-import com.wynntils.features.debug.KeyBindTestFeature;
 import com.wynntils.features.debug.PacketDebuggerFeature;
+import com.wynntils.features.internal.FixPacketBugsFeature;
+import com.wynntils.features.internal.LootrunFeature;
+import com.wynntils.features.user.DialogueOptionOverrideFeature;
+import com.wynntils.features.user.EmeraldPouchHotkeyFeature;
+import com.wynntils.features.user.GammabrightFeature;
+import com.wynntils.features.user.HealthPotionBlockerFeature;
+import com.wynntils.features.user.IngredientPouchHotkeyFeature;
+import com.wynntils.features.user.ItemGuessFeature;
+import com.wynntils.features.user.ItemHighlightFeature;
+import com.wynntils.features.user.ItemScreenshotFeature;
+import com.wynntils.features.user.ItemStatInfoFeature;
+import com.wynntils.features.user.MountHorseHotkeyFeature;
+import com.wynntils.features.user.MythicBlockerFeature;
+import com.wynntils.features.user.PlayerGhostTransparencyFeature;
+import com.wynntils.features.user.SoulPointTimerFeature;
+import com.wynntils.features.user.WynncraftButtonFeature;
 import com.wynntils.mc.event.ClientTickEvent;
 import com.wynntils.mc.event.RenderEvent;
 import com.wynntils.mc.utils.CrashReportManager;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wc.utils.WynnUtils;
-import java.util.HashMap;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 /** Loads {@link Feature}s */
 public class FeatureRegistry {
-    private static final Map<Class<? extends Feature>, Feature> FEATURES = new HashMap<>();
-    private static final Map<Class<? extends Overlay>, Overlay> OVERLAYS = new HashMap<>();
+    private static final List<Feature> FEATURES = new ArrayList<>();
+    private static final List<Overlay> OVERLAYS = new ArrayList<>();
 
     public static void registerFeature(Feature feature) {
         if (feature instanceof Overlay overlay) {
-            OVERLAYS.put(overlay.getClass(), overlay);
+            OVERLAYS.add(overlay);
         }
 
-        FEATURES.put(feature.getClass(), feature);
+        FEATURES.add(feature);
+
+        initializeFeature(feature);
```

Comment:
This may be better off as automatically called from something like i.e. the constructor

Reply:
Initialization? There might be a point of having an init step done at a later point than construction. The constructor is quite limited in how you access fields and internal methods etc, so sometimes it's hard to write proper code there.